### ISSUE TITLE
Some routes fail to authenticate with OIDC

### ISSUE THREAD

**Issue by azmeuk** [pulsejet/nextcloud-oidc-login#190]
Even with `oidc_login_password_authentication` enabled, a simple request on the [nextcloud news](https://github.com/nextcloud/news) API always fails with OIDC credentials, where it succeeds with an *app password*.

```bash
$ # where `xxxxxxx` is `base64($user.":".$password)` and $password is the password from an OIDC account
$ curl -H "Authorization: Basic xxxxxxxx" "https://nextcloud.tld/index.php/apps/news/api/v1-3/version"
{"message":""}

$ # where `xxxxxxx` is `base64($user.":".$password)` and $password is an app password generated in nextcloud
$ curl -H "Authorization: Basic xxxxxxxx" "https://nextcloud.tld/index.php/apps/news/api/v1-3/version"
{"version":"18.3.0-beta1"}
```

The endpoint callgraph is:
- [index.php: handleRequest](https://github.com/nextcloud/server/blob/baa373e40a9bae62d9a7adf1e29bf0f843fcda59/index.php#L36)
- [lib/base.php: handleLogin](https://github.com/nextcloud/server/blob/baa373e40a9bae62d9a7adf1e29bf0f843fcda59/lib/base.php#L1019)
- [lib/base.php: tryBasicAuthLogin](https://github.com/nextcloud/server/blob/baa373e40a9bae62d9a7adf1e29bf0f843fcda59/lib/base.php#L1081)
- [lib/private/User/Session.php: logClientIn](https://github.com/nextcloud/server/blob/baa373e40a9bae62d9a7adf1e29bf0f843fcda59/lib/private/User/Session.php#L575)
- [lib/private/User/Session.php: login](https://github.com/nextcloud/server/blob/baa373e40a9bae62d9a7adf1e29bf0f843fcda59/lib/private/User/Session.php#L450)
- [lib/private/User/Session.php: loginWithPassword](https://github.com/nextcloud/server/blob/baa373e40a9bae62d9a7adf1e29bf0f843fcda59/lib/private/User/Session.php#L355)
- [lib/private/User/Session.php: checkPasswordNoLogging](https://github.com/nextcloud/server/blob/baa373e40a9bae62d9a7adf1e29bf0f843fcda59/lib/private/User/Session.php#L610)
- that fails because the only [registered backend](https://github.com/nextcloud/server/blob/baa373e40a9bae62d9a7adf1e29bf0f843fcda59/lib/private/User/Manager.php#L256) is `OC\User\Database`

I suppose the solution to fix this would be to register nextcloud-oidc-login as a user backend.

@sirkrypt0 any thoughts?

**Comment by pulsejet** [comment]
Nice work debugging this, and your fix seems on point. Implementing this might change the login process somewhat (not sure).

I'm curious though why the news app doesn't support token authentication. Unless I'm missing something here, this is more of a problem of theirs than us. WebDAV seems like a one-off case where you might want to use oidc password login (ideally you'd use an app password here too).

**Comment by azmeuk** [comment]
I only had tested `Basic` authentication with `oidc_login_password_authentication`, but indeed *news* does not seem to support `Bearer` authentication:

```bash
$ # where `xxxxxxx` is a OIDC access token
$ curl -H "Authorization: Bearer xxxxxxx" "https://nextcloud.tld/index.php/apps/news/api/v1-3/version"
{"message":"CORS requires basic auth"}
```

> Implementing this might change the login process somewhat (not sure).

Yeah. I fear this would be a lot of work.

> Unless I'm missing something here, this is more of a problem of theirs than us.

Why that?
Sure, news could support `Bearer` authentication, and this one does not depend on us.
But we also fail to register as a valid `Basic` backend on the route they use. This feels like a problem for us. What do you think?

**Comment by sirkrypt0** [comment]
Yes, implementing a user backend is probably the best way to go here, similar to what the "official" [`user_oidc`](https://github.com/nextcloud/user_oidc) plugin does I think.
